### PR TITLE
Updated Local file upload prompt

### PR DESCRIPTION
**Pull request by muskangarg98**
Updated the Local File Uploader UI and added some checks.
Added a file picker so that the user can decide which drive folder they want the uploaded file to be stored to.

1. New Local File Upload From
Screenshot 2020-07-06 at 12 47 02 PM

2. Check if the uploaded file and local path mimetype matches or not
Screenshot 2020-07-06 at 12 55 44 PM

3. Success message
Screenshot 2020-07-06 at 12 56 49 PM

### CODE REVIEW

File: FileMapper/fileUploadForm_script.html [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+   *
+   * @param {String} extension The extension of the local path
+   * @return {String} MimeType of the file based on it's extension
+   */
+  function getMimetypeFromExtension(extension) {
+    if (extension === "xls" || extension === "xlsx" || extension === "csv")
+      return 'GOOGLE_SHEETS';
+
+    if (extension === "doc" || extension === "docx" || extension === "pdf" ||
+        extension === "txt")
```
Comment (paneendra-google) [inline FileMapper/fileUploadForm_script.html]: not sure if we want to convert text file to docs, we should not turn the conversion on for text files.

File: FileMapper/fileUploadForm_script.html [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+   */
+  function checkIfExtensionMatches(uploadedFile) {
+    var localPath = document.getElementById("localpathinput").value;
+
+    var uploadedFileExtension = getExtensionFromLocalPath(uploadedFile);
+    var localPathExtension = getExtensionFromLocalPath(localPath);
+
+    if (getMimetypeFromExtension(localPathExtension) ===
+        getMimetypeFromExtension(uploadedFileExtension)) {
+      showUploadedFileMessage("File type matches to the local file type", true);
```
Comment (paneendra-google) [inline FileMapper/fileUploadForm_script.html]: to the -> the, also in the message below

**Review by paneendra-google — CHANGES_REQUESTED** [review]

File: FileMapper/fileUploadForm_script.html [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  }
+
+  /**
+   * Function to return the MimeType based on the file extension
+   *
+   * @param {String} extension The extension of the local path
+   * @return {String} MimeType of the file based on it's extension
+   */
+  function getMimetypeFromExtension(extension) {
+    if (extension === "xls" || extension === "xlsx" || extension === "csv")
```
Comment (paneendra-google) [inline FileMapper/fileUploadForm_script.html]: Would prefer if we create arrays like

var sheet_types = ["xls", "xlsx", xlsm", "csv", "tsv"] and do "if (extension in sheet_types)" instead.

Also should add xlsm and TSV here and docm and pptm in the other two

**Review by paneendra-google — CHANGES_REQUESTED** [review]

File: FileMapper/fileUploadForm_script.html (hunk 1 above)
Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: So, should we keep the mimetype for text files as PLAIN_TEXT?

File: FileMapper/fileUploadForm_script.html (hunk 3 above)
Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: "if (extension in sheet_types)" doesn't work with arrays, instead we can use "sheet_types.indexOf(extension)" method, but this too will take linear time. 
Should I create arrays then?

File: FileMapper/fileUploadForm_script.html (hunk 2 above)
Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: Fixed

File: FileMapper/fileUploadForm_script.html (hunk 3 above)
Comment (paneendra-google) [reply FileMapper/fileUploadForm_script.html]: Yes, it's not about the time complexity, I think code will be easier to read and adding more extensions for a particular type is much more simpler

File: FileMapper/fileUploadForm_script.html (hunk 1 above)
Comment (paneendra-google) [reply FileMapper/fileUploadForm_script.html]: Yes, please

**Review by paneendra-google — CHANGES_REQUESTED** [review]

File: FileMapper/fileUploadForm_script.html (hunk 3 above)
Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: Yeah right. I will change it.

Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: Done

File: FileMapper/fileUploadForm_css.html [hunk 4]
```diff
@@ -0,0 +1,121 @@
+<!--
```
Comment (paneendra-google) [inline FileMapper/fileUploadForm_css.html:1]: file name file_upload_form_css.html, similarly for others

File: FileMapper/fileUploadForm_script.html [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+  }
+
+  /**
+   * Function to return the MimeType based on the file extension
+   *
+   * @param {String} extension The extension of the local path
+   * @return {String} MimeType of the file based on it's extension
+   */
+  function getMimetypeFromExtension(extension) {
+    var sheets_extensions = ["xls", "xlsx", "csv", "xlsm", "tsv"];
```
Comment (paneendra-google) [inline FileMapper/fileUploadForm_script.html]: I guess this change was done in another PR? once that is merged, sync and re-open this PR

**Review by paneendra-google — CHANGES_REQUESTED** [review]

File: FileMapper/fileUploadForm_css.html (hunk 4 above)
Comment (muskangarg98) [reply FileMapper/fileUploadForm_css.html:1]: Okay, will update all file names once all the PRs are merged.

File: FileMapper/fileUploadForm_script.html (hunk 5 above)
Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: This is not the same file, this is on the client-side, that was on the server-side. We can make a call to that same function here but then it will be a call to the server which might take some extra time. What do you suggest?

Comment (paneendra-google) [reply FileMapper/fileUploadForm_script.html]: We can have it as a separate js file which could be included both on the server and the client?

Comment (muskangarg98) [reply FileMapper/fileUploadForm_script.html]: I couldn't find a way to include a js file on the client-side code in appscript. 
The javascript code can be used from a separate html file using templating but there is no way to use it from a javascript file.

**Review by paneendra-google — DISMISSED** [review]

**Comment by muskangarg98** [comment]
I have made a separate HTML script file which contains the "getMimetypeFromExtension" function and is shared across File Picker and the Local File Uploader. Now, we don't need to have repetition atleast on the client-side.

**Review by paneendra-google — APPROVED** [review]
